department-of-veterans-affairs/va.gov-cms
# VACMS-11388 Add public version of accessCheck.

This mimics what is created in
docroot/modules/contrib/workbench_menu_access/src/WorkbenchMenuLinkContentAccessControlHandler.php

## Description
Closes #11388

## Testing done
## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Go to /admin/structure/menu/item/723/edit (previously inaccessible due to white-screen)
   - [ ] Validate that you can see the menu edit form
   - [ ] Notice that the parent is empty
   - [ ] Save the menu item (without changing anything)
   - [ ] Hit the back button to edit the item again, notice that the parent is now filled in.
2. Go to /admin/structure/menu/item/719/edit (this is one that worked before
   - [ ] Validate that you can see the menu edit form
   - [ ] validate that it has a parent set.
   - [ ] Save the menu item (without changing anything)
   - [ ] Hit the back button to edit the item again, notice that the parent is still filled in.


### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
